src/core/experience/service/experience.service.ts

```diff
@@ -72,6 +80,30 @@ export class ExperienceService {
     return;
   }
 
+  async removeMyExperience(
+    id: number,
+    user: UserActiveInterface,
+  ): Promise<void> {
+    const experience = await this.experienceRepository.findOneBy({ id });
+    if (!experience) throw new NotFoundException(EXPERIENCE_NOT_FOUND);
+
+    if (experience.profileId !== user.id)
+      throw new ForbiddenException(YOU_NOT_HAVE_PERMISSION_MESSAGE);
+
+    try {
+      const experienceToRemove = await this.experienceRepository.softDelete({
+        profileId: user.id,
+        id: id,
+      });
+      if (experienceToRemove.affected === 0) {
+        throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
```

Comment:
Bella lógica de negocios ❤.
Un último fix
```suggestion
      const experienceRemoved = await this.experienceRepository.softDelete({
        profileId: user.id,
        id: id,
      });
      if (experienceRemoved.affected === 0) {
        throw new InternalServerErrorException(INTERNAL_SERVER_ERROR);
```
Es una experiencia que ya se elimino, en tiempo pasado, no se va a eliminar